smoke-test.sh

```diff
@@ -7,6 +7,7 @@ VERSION_PREFIX=2.1
 # See https://github.com/dotnet/source-build/issues/579, this version
 # needs to be compatible with the runtime produced from source-build
 DEV_CERTS_VERSION_DEFAULT=2.1.0-rtm-30762
+__ROOT_REPO=$(cat "$SCRIPT_ROOT/bin/obj/rootrepo.txt" | sed 's/\r$//') # remove CR if mounted repo on Windows drive
```

Comment:
I don't think this is the cleanest way to pass information but I cannot think of another nice way. @dagood @crummel if we end up needing to pass in more information into this test script perhaps we make it hook into our msbuild projects so that we can share.

Reply:
Hmm, good point. Having a rootrepo.txt file means that if a user chooses to manually pass in a root repo, smoke-test will have "nice" results. But the user should be able to understand what the error message means when smoke-test can't find the CLI tarball. (If not, we can improve that error message.)

This could actually be fixed pretty quick: have a `/t:RunSmokeTest` that checks what RootRepo ends up being, and only `Exec`s `smoke-test.sh` if RootRepo is enough. Then, instead of calling smoke-test.sh in CI, call `./build.sh /t:RunSmokeTest`.